### ISSUE TITLE
Task :react-native-navigation:compileReactNative57_5DebugJavaWithJavac FAILED

### ISSUE THREAD

**Issue by bunny100** [wix/react-native-navigation#5273]
### Issue Description

Getting this error while running 'react-native run-android'
![image](https://user-images.githubusercontent.com/46201508/60914612-5327d880-a2a3-11e9-90fd-cf62f25941d6.png)

Android Studio compiler output:
![image](https://user-images.githubusercontent.com/46201508/60914746-93875680-a2a3-11e9-91c4-1df89ccd9108.png)

running with 'npm run android':
![image](https://user-images.githubusercontent.com/46201508/60914834-ccbfc680-a2a3-11e9-9006-42bf7ee20de2.png)

running with 'gradlew app:assembleDebug and gradlew installDegug':
![image](https://user-images.githubusercontent.com/46201508/60915043-4c4d9580-a2a4-11e9-9f5c-d10f35de4757.png)

I have tried almost everything and not able to set things up, Please someone help.

**Comment by igortupini** [comment]
Have you followed installation steps **5** and/or **5.2**  correctly? Those are described here:
https://wix.github.io/react-native-navigation/#/docs/Installing?id=android

If you already have done that, can you post here your `android/app/build.gradle` **defaultConfigs** `android/build.gradle` **subprojects** files?

What's your RN version?

**Comment by bunny100** [comment]
default configs:

def useIntlJsc = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "com.project"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative57_5"
        versionCode 1
        versionName "1.0"
    }

![image](https://user-images.githubusercontent.com/46201508/60922455-2df09580-a2b6-11e9-8c01-1f3f4cf6aad1.png)



subprojects file:

   subprojects { subproject ->
            afterEvaluate {
                    if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
                            android {
                                    variantFilter { variant ->
                                            def names = variant.flavors*.name
                                            if (names.contains("reactNative51") || names.contains("reactNative55")) {
                                                    setIgnore(true)
                                                }
                                        }
                                }
                        }
                }
        }
![image](https://user-images.githubusercontent.com/46201508/60922472-3ba61b00-a2b6-11e9-8e88-57de12c8b6ef.png)

I'm using RN 0.60.0
minsdk = 19
targetsdk = 28

![image](https://user-images.githubusercontent.com/46201508/60922576-7e67f300-a2b6-11e9-873a-6d2aaeb4453c.png)

![image](https://user-images.githubusercontent.com/46201508/60922643-9e97b200-a2b6-11e9-8fc6-ad3113bf0dda.png)

**Comment by igortupini** [comment]
I don't believe that your currently RNN version supports **React Native** `0.60`.
If you want to use RNN I would recommend downgrade to React Native `0.59.10`. 
You can also check #5228 issue about RNN V3 that will support RN `0.60`

**Comment by bunny100** [comment]
that can be the very problem, I'm trying v3 now.

**Comment by bunny100** [comment]
problem is still there bro. I think, I should downgrade.
They must release a stable version of RNN for RN 0.6.0, its frustrating that I haven't got any solution for 3 days. well thanks for your support.

**Comment by igortupini** [comment]
> problem is still there bro. I think, I should downgrade.
> They must release a stable version of RNN for RN 0.6.0, its frustrating that I haven't got any solution for 3 days. well thanks for your support.

V3 still on alpha, so you should expect bugs and other issues.
RN 0.60 was released less than 2 weeks ago, and has changed quite a few things so most libraries still undergoing upgrade process.

Unless you have some real problem that only RN 0.60 will help you I really suggest waiting a little more before upgrading.

**Comment by jinshin1013** [comment]
Try adding

`missingDimensionStrategy "RNN.reactNativeVersion", "reactNative60"`

```
subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion

                variantFilter { variant ->
                    def names = variant.flavors*.name
                    if (
                        names.contains("reactNative51") || 
                        names.contains("reactNative55") || 
                        names.contains("reactNative56") ||
                        names.contains("reactNative57") ||
                        names.contains("reactNative57_5") 
                    ) {
                        setIgnore(true)
                    }
                }
            }
        }
    }
}
```

**Comment by Frank17** [comment]
This problem can be solved after removing androidX supporting in gradle.properties: 

```
# Project-wide Gradle settings.

# IDE (e.g. Android Studio) users:
# Gradle settings configured through the IDE *will override*
# any settings specified in this file.

# For more details on how to configure your build environment visit
# http://www.gradle.org/docs/current/userguide/build_environment.html

# Specifies the JVM arguments used for the daemon process.
# The setting is particularly useful for tweaking memory settings.
# Default value: -Xmx10248m -XX:MaxPermSize=256m
# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8

# When configured, Gradle will run in incubating parallel mode.
# This option should only be used with decoupled projects. More details, visit
# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
# org.gradle.parallel=true

#android.useAndroidX=true
#android.enableJetifier=true
```
just remove the two lines : 
```
android.useAndroidX=true
#android.enableJetifier=true
```

**Comment by amdrade** [comment]
I removed AndroidX, it did not work for me :(

![AwesomeScreenshot-Desktop-screenshot-2019-07-11-12-07-18](https://user-images.githubusercontent.com/5581309/61063853-12c97700-a3d7-11e9-8d81-4bbc1c6ef21a.png)

**Comment by Frank17** [comment]
Have you config the `defaultConfig` in the build.gradle, may be you should ignore more variants like @jinshin1013 ? And the build variants need to be `57_5` @amdrade

**Comment by jinshin1013** [comment]
@Frank17, if you look at V3 commits guyca added a new installation instruction which tells us to add `0.60.0` as the build variants if you are using `react-native@0.60.`. Also shouldn't be removing below I reckon:

```
android.useAndroidX=true
#android.enableJetifier=true
```

**Comment by Frank17** [comment]
@jinshin1013 Yeah, I think it will support AndroidX, but if I install react-native-navigation by npm, I can't  get the lib like `v3`, so I removed it.

**Comment by Andarius** [comment]
> Try adding
> 
> `missingDimensionStrategy "RNN.reactNativeVersion", "reactNative60"`
> 
> ```
> subprojects { subproject ->
>     afterEvaluate {
>         if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
>             android {
>                 compileSdkVersion rootProject.ext.compileSdkVersion
>                 buildToolsVersion rootProject.ext.buildToolsVersion
> 
>                 variantFilter { variant ->
>                     def names = variant.flavors*.name
>                     if (
>                         names.contains("reactNative51") || 
>                         names.contains("reactNative55") || 
>                         names.contains("reactNative56") ||
>                         names.contains("reactNative57") ||
>                         names.contains("reactNative57_5") 
>                     ) {
>                         setIgnore(true)
>                     }
>                 }
>             }
>         }
>     }
> }
> ```

For brevity you can do something like : ` if(variant.flavors*.name.any{it.contains('reactNative5')}){ setIgnore(true) }`

**Comment by cschaffr** [comment]
most alpha versions don't include reactNative60 directory. Seems like the only one which works on android is v3 alpha.4
I get the same error messages with newer/older alpha versions.

**Comment by iShankarG** [comment]
> Try adding
> 
> `missingDimensionStrategy "RNN.reactNativeVersion", "reactNative60"`
> 
> ```
> subprojects { subproject ->
>     afterEvaluate {
>         if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
>             android {
>                 compileSdkVersion rootProject.ext.compileSdkVersion
>                 buildToolsVersion rootProject.ext.buildToolsVersion
> 
>                 variantFilter { variant ->
>                     def names = variant.flavors*.name
>                     if (
>                         names.contains("reactNative51") || 
>                         names.contains("reactNative55") || 
>                         names.contains("reactNative56") ||
>                         names.contains("reactNative57") ||
>                         names.contains("reactNative57_5") 
>                     ) {
>                         setIgnore(true)
>                     }
>                 }
>             }
>         }
>     }
> }
> ```

not getting error by this, but Metro Bundler not starting and manually started Metro Bundler is not connecting to app.

**Comment by loga4** [comment]
> not getting error with this, but Metro Bundler not starting and manually started Metro Bundler is not connecting to app.

downgrade node to 12.4

about subj, not working:
`Task :react-native-navigation:compileReactNative60DebugJavaWithJavac FAILED`

> `missingDimensionStrategy "RNN.reactNativeVersion", "reactNative60"`
> 
> ```
> subprojects { subproject ->
>     afterEvaluate {
>         if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
>             android {
>                 compileSdkVersion rootProject.ext.compileSdkVersion
>                 buildToolsVersion rootProject.ext.buildToolsVersion
> 
>                 variantFilter { variant ->
>                     def names = variant.flavors*.name
>                     if (
>                         names.contains("reactNative51") || 
>                         names.contains("reactNative55") || 
>                         names.contains("reactNative56") ||
>                         names.contains("reactNative57") ||
>                         names.contains("reactNative57_5") 
>                     ) {
>                         setIgnore(true)
>                     }
>                 }
>             }
>         }
>     }
> }
> ```

**Comment by Tomasamado26** [comment]
The first bunch of errors you got @bunny100 you can fix them by going to node_modules/react-native-navigation and changing the imports to the AndroidX ones. For example: `import android.support.v4.view.ViewPager;` to `import androidx.viewpager.widget.ViewPager;`
Doing this I managed to get the number of errors down. I still haven´t fixed all of the errors, but here are a lot of imports ones.

Its kinda boring to have to fix so many of those errors by hand but this should be fixed on the new version of RNN.

**Comment by filipepiresg** [comment]
You can try using jetify.

First, install the jetifier using the command `yarn add -D jetifier` or `npm install -D jetifier` and then you can run the command `npx` jetify` .

For me, that solved this problem

**Comment by Tomasamado26** [comment]
I did that but it didn´t solve anything, still got those errors.

**Comment by bneigher** [comment]
@cschaffr alpha 8 was working for me, now 11 isn't

**Comment by jkol36** [comment]
did anyone find a fix for this? Where do i put  `subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion rootProject.ext.compileSdkVersion
                buildToolsVersion rootProject.ext.buildToolsVersion

                variantFilter { variant ->
                    def names = variant.flavors*.name
                    if (
                        names.contains("reactNative51") || 
                        names.contains("reactNative55") || 
                        names.contains("reactNative56") ||
                        names.contains("reactNative57") ||
                        names.contains("reactNative57_5") 
                    ) {
                        setIgnore(true)
                    }
                }
            }
        }
    }
}`

**Comment by sana948** [comment]
The variantFilter block logic goes inside android/build.gradle file

The `missingDimensionStrategy "RNN.reactNativeVersion", "reactNative60"` inside android/app/build.gradle file

**Comment by FinnGu** [comment]
I know this issue is old, but I just had the same problem using react-native 0.68.7 and react-native-navigation 7.39.2. Which version should I use as `missingDimensionStrategy`? And which versions do I need to ignore in my `build.gradle` file?
